src/unittest_templates/api.py

```diff
@@ -24,9 +24,9 @@ class GenericTestCase(Generic[T], unittest.TestCase):
     def setUp(self) -> None:
         """Set up the generic testing method."""
         self.pre_setup_hook()
-        kwargs = self.kwargs or {}
-        kwargs = self._pre_instantiation_hook(kwargs=dict(kwargs))
-        self.instance = self.cls(**kwargs)  # type: ignore
+        kwargs = dict(self.kwargs or {})
+        self.kwargs = self._pre_instantiation_hook(kwargs=kwargs)
```

Comment:
maybe this would be better in `setUpClass`? Does the _pre_instantiation_hook function ever need to use instance-level information?

Reply:
otherwise the kwargs get clobbered on the first test. you could also do
```python
self.instance_kwargs = self._pre_instantiation_hook(kwargs=kwargs)
``` 
so you know what the difference between the initial and the processed ones are if it's potentially different for each run of setUp

Reply:
Now, the `kwargs` get only modified in the `_pre_instantiation_hook`, which is called *before* the class is initialized. So it *should* be save to do it in the `setUpClass` as well. However, I prefer the second solution,

Reply:
the `setUp()` is run before each test function, so as it is, the `_pre_instantation_hook()` gets run once for each test function, meaning that this change would result in modifying the kwargs over and over again. the `setUpClass()` is run once as the class is instantiated